src/lib/headline_filter.js

```diff
@@ -24,6 +198,10 @@ export const isMatch = (filterExpr) => (header) => {
   const filterProps = filterExpr
     .filter(filterFilter('property', false))
     .map((x) => [x.property, x.words]);
+  const filterField = filterExpr.filter(filterFilter('field', false));
+  const filterClock = filterField.filter((f) => f.field.type === 'clock').map(timeFilter);
```

Comment:
Sorry, I didn't fully analyze your new code. Just a question: Have you considered that filter terms can be negated with "-"? I don't remember if that applies for all filter terms or just for the existing ones.

Does it make sense for date queries? Not necessarily, I think. But in that case, maybe the search grammar must be changed.

Reply:
Something like `clock:2020 -clock:202004` meaning all year without april makes sense. I don't know if it is that useful though.
Seems like I just ignore negated timerange-searchqueries at the moment. That's not expected behaviour. Since for now all filters are negateable it would make sense to do it here too.

Reply:
But don't start too complex ;) you can as well just throw a "parse exception" or something to tell the user that negating this filter is not yet supported. Implement the important (and practical) features first.

Reply:
I throw. 

I'm not quite sure if the logic should be the same as in text search.
In text search if a header text matches a filter and an exclusion filter, it is not shown.
If a header has a timestamp that matches a filter and an exclusion filter it should also not be shown.
But what if a header has one timestamp that matches a filter and another timestamp that matches an exclusion filter? I think it should be included but it's not clear cut.
=> negated timeranges would be easy if we only allowed for one time filter in a search string. Since there can be an arbitrary amount of filters, they are tricky. I won't touch them for now,

Reply:
That even a human can't easily answer :D My feeling is that exclusion should have "the last word", i.e. exclude headline if an exclude filter matches.

But not allowing these things is good as well. It also makes the whole thing easier to understand.

Reply:
"Exclusion" always trumps inclusion when searching. Think of it this way: When searching without a term, the default is to show everything. As soon as you introduce a constraint, it will remove terms from the resultset. They cannot be brought back by further inclusions.

Not really a search engine, but the behavior is the same:

```
$ cat > great_editors
emacs
vim
$ cat great_editors | grep -v vim | grep vim | wc -l
0
```